### PR TITLE
fix: handle socket file detection on Windows 

### PR DESCRIPTION
**Pull request by fakecore**
Update socket file detection logic to use os.Stat as per upstream Go fix for https://github.com/golang/go/issues/33357. This resolves the issue where socket files could not be properly identified on Windows systems.

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Update to use os.Stat instead of os.Lstat when deleting the device-plugins socket file, rather than deleting all files in the /var/lib/kubelet/device-plugins/ directory.

#### Which issue(s) this PR fixes:

related issue: https://github.com/kubernetes/kubernetes/issues/97554
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

**Comment by hakman** [comment]
@fakecore I don't see commit https://github.com/golang/go/commit/628b1015b972eabcc0a678ab69a74601239c40a4 cherry-picked to any of the current Go release branches.
I think merging this PR should wait until Kubernetes is built using a Go version containing the above mentioned commit.

**Comment by ffromani** [comment]
> @fakecore I don't see commit [golang/go@628b101](https://github.com/golang/go/commit/628b1015b972eabcc0a678ab69a74601239c40a4) cherry-picked to any of the current Go release branches. I think merging this PR should wait until Kubernetes is built using a Go version containing the above mentioned commit.

I agree. Looks like a nice fix, but we need to make sure the dependencies are satisfied. I'll check what's the best triage labeling for this kind of state. Holding meanwhile.

**Comment by ffromani** [comment]
/hold

per https://github.com/kubernetes/kubernetes/pull/123819#issuecomment-1987113850

**Comment by fakecore** [comment]
@cblecker @hakman @ffromani 
Thank you for the review. It was my mistake, I should have checked the release status of Go and the dependencies of Kubernetes. I will be more cautious next time.

**Comment by ffromani** [comment]
> @cblecker @hakman @ffromani Thank you for the review. It was my mistake, I should have checked the release status of Go and the dependencies of Kubernetes. I will be more cautious next time.

It's fine, no problem at all! It's just this PR unfortunately will have to wait for a while before it can be merged. Once the golang dep is updated, it should go in smoothly though.

**Comment by ffromani** [comment]
> hi all, I noticed current kubernets build version is go1.22.4 this patch release with go1.22.1 can we merge this ?

I'd bump go.mod to 1.22.1 at least, or somehow require at least go 1.22.1 before to merge

**Comment by fakecore** [comment]
> > hi all, I noticed current kubernets build version is go1.22.4 this patch release with go1.22.1 can we merge this ?
> 
> I'd bump go.mod to 1.22.1 at least, or somehow require at least go 1.22.1 before to merge

Could we use the version specified in `.go-version` instead?

**Comment by ffromani** [comment]
> > > hi all, I noticed current kubernets build version is go1.22.4 this patch release with go1.22.1 can we merge this ?
> > 
> > 
> > I'd bump go.mod to 1.22.1 at least, or somehow require at least go 1.22.1 before to merge
> 
> Could we use the version specified in `.go-version` instead?

right, forgot about this. I think it's fine? cc @cpanato to crosscheck

**Comment by ffromani** [comment]
/hold
/lgtm

AFAIU master branch is already on golang >= 1.22.4, so this seems safe to move forward

**Comment by fakecore** [comment]
@ffromani Should I do anything to expedite the merging of this PR? It's been hanging there for a while.

**Comment by ffromani** [comment]
> @ffromani Should I do anything to expedite the merging of this PR? It's been hanging there for a while.

we need approval review, tagging few ppl which can help @klueska @mrunalp @dchen1107

**Comment by fakecore** [comment]
> > @ffromani Should I do anything to expedite the merging of this PR? It's been hanging there for a while.
> 
> we need approval review, tagging few ppl which can help @klueska @mrunalp @dchen1107

I have no idea who the maintainer is for this node. Thanks for your help!

**Review by mrunalp — APPROVED** [review]

**Comment by ffromani** [comment]
/hold cancel

no longer relevant (xref: https://github.com/kubernetes/kubernetes/pull/123819#issuecomment-2210388500)

**Pull request merged**
